ui/src/criteria/attribute.tsx

```diff
@@ -81,23 +133,145 @@ function AttributeEdit(props: AttributeEditProps) {
+    const maxBound = integerBoundsHint?.max || 0; // although we already know that min and max is valid
+
+    //two sets of values are needed due to the input box and slider is isolated.
+    const [minInputValue, setMinInputValue] = useState<number>(minBound);
+    const [maxInputValue, setMaxInputValue] = useState<number>(maxBound);
+    const [minValue, setMinValue] = useState<number>(minBound);
+    const [maxValue, setMaxValue] = useState<number>(maxBound);
+
+    const updateValue = (newMin: number, newMax: number) => {
+      setMinValue(newMin);
+      setMaxValue(newMax);
+      props.dispatchFn(
+        produce(props.data, (data) => {
+          data.min = newMin;
+          data.max = newMax;
+        })
+      );
+    };
+
+    const handleChange = (event: Event, newValue: number | number[]) => {
+      newValue = newValue as number[];
+      const [newMin, newMax] = newValue;
+      setMinInputValue(newMin);
+      setMaxInputValue(newMax);
+      updateValue(newMin, newMax);
+    };
+
+    const handleLeftInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      let newMin =
+        event.target.value === "" ? minBound : Number(event.target.value);
+      setMinInputValue(newMin);
+
+      newMin = Math.max(minBound, newMin);
+      newMin = Math.min(maxValue, newMin);
+      updateValue(newMin, maxValue);
+    };
+    const handleLeftInputBlur = () => {
+      let newMin = minInputValue;
+      newMin = Math.max(newMin, minBound);
+      newMin = Math.min(newMin, maxValue);
+      setMinInputValue(newMin);
+    };
+
+    const handleRightInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      let newMax =
+        event.target.value === "" ? maxBound : Number(event.target.value);
+      setMaxInputValue(newMax);
+
+      newMax = Math.max(minValue, newMax);
+      newMax = Math.min(newMax, maxBound);
+      updateValue(minValue, newMax);
+    };
+    const handleRightInputBlur = () => {
+      let newMax = maxInputValue;
+      newMax = Math.max(minValue, newMax);
+      newMax = Math.min(newMax, maxBound);
+      setMaxInputValue(newMax);
+    };
+
+    return (
+      <Box sx={{ width: "30%", minWidth: 100 }}>
+        <Grid container spacing={2} alignItems="center">
+          <Grid item>
+            <Input
+              value={minInputValue}
+              size="small"
+              onChange={handleLeftInputChange}
+              onBlur={handleLeftInputBlur}
+              inputProps={{
+                step: ((maxBound - minBound) / 20) | 0,
+                min: minBound,
+                max: maxBound,
+                type: "number",
+                "aria-labelledby": "input-slider",
+              }}
+            />
+          </Grid>
+          <Grid item xs>
+            <Slider
+              getAriaLabel={() => "Year at birth range"}
```

Comment:
This can't be hardcoded. It needs to come from the from the `CriteriaConfig`, probably `` `${config.title} Range` ``.

Reply:
I find I made a mistake for this property "getAriaLabel" is not used to show the title/label of the slider. It required an index. Seems like we do not really need it. I just delete it

Reply:
getAriaLabel and ARIA properties in general are for interacting with screen readers, keyboard input, and other accessibility related things. I'm not sure exactly what issue you're running into but for now it's fine to just delete it.